### PR TITLE
Make collision module doable for RZ

### PR DESCRIPTION
**Pull request by Yin-YinjianZhao**
This PR attempts to make the collision module doable for 2D-RZ coordinate under the simplest consideration at the moment. i.e. only the cell volume `dV` is changed for 2D-RZ.
After figuring out the cell index `i` in the `r` direction, the cell volume can be calculated from:
`dV = { pi * [(i+1)*dr]^2 - pi * (i*dr)^2 } * dz = pi * (2i+1) * dr^2 * dz`.

@RemiLehe Do you think this is correct?

### CODE REVIEW

**Comment by Yin-YinjianZhao** [comment]
@atmyers Could you please have a look at the code and see if I'm correct from the coding side? Thanks!

**Comment by RemiLehe** [comment]
@dpgrote Do you have any suggestion for the calculation of the cell volume in RZ (see the above question by @Yin-YinjianZhao)?
@Yin-YinjianZhao Could you also add an automated test, for collisions in RZ, as part of this PR?

**Comment by dpgrote** [comment]
Hi @Yin-YinjianZhao Your equation for dV looks fine for calculating the volume for a cell centered grid. However, what is the ordering of i_cell? Does it loop over r or z first? Your equation for ri looks like it is calculating iz. Should it be ri = i_cell%nr, if i_cell is looping over r first?

**Comment by Yin-YinjianZhao** [comment]
> Hi @Yin-YinjianZhao Your equation for dV looks fine for calculating the volume for a cell centered grid. However, what is the ordering of i_cell? Does it loop over r or z first? Your equation for ri looks like it is calculating iz. Should it be ri = i_cell%nr, if i_cell is looping over r first?

Thanks Dave for your check! I believe `i_cell` is looping over `z` first. I renamed `nr` to `nz`.
So, the formula should be `ri = (i_cell - i_cell%nz) / nz;`
And we can find `nz` from:
```
        Box const& cbx = mfi.tilebox(IntVect::TheZeroVector()); //Cell-centered box
        const auto lo = lbound(cbx);
        const auto hi = ubound(cbx);
        int nz = hi.y-lo.y+1;
```
Am I right? @atmyers

File: Source/Particles/Collision/CollisionType.cpp [hunk 1]
```diff
@@ -156,6 +159,11 @@ void CollisionType::doCoulombCollisionsWithinTile
                     ShuffleFisherYates(
                         indices_1, cell_start_1, cell_half_1 );
 
+#if defined WARPX_DIM_RZ
+                    int ri = (i_cell - i_cell%nz) / nz;
```
Comment (dpgrote) [inline Source/Particles/Collision/CollisionType.cpp:163]: This makes more sense now, with `nz`. I might add a comment that `i_cell` varies most rapidly in z as calculated by ParticleBins.build.

**Comment by Yin-YinjianZhao** [comment]
This PR should work, since it is tested using an emittance growth case.
As you can see from the plot below, where two curves one in 3D another in RZ pretty much match.
However, this test is not suitable to be made as an automated test, since it runs for hours.
e

**Review by RemiLehe — APPROVED** [review]

**Pull request merged**